### PR TITLE
feature: submissions dashboard

### PR DESCRIPTION
**Pull request by jwsmith24**
With these changes, Palette is now able to fetch all submissions for an assignment from Canvas, organize the submissions into groups, and display each group (along with grading progress) in a dashboard-style interface. 

If submissions don't have a group association, they're added to a group that's just called "No Group". I'm thinking eventually, we could implement an alternative dashboard view depending on if the assignment is a group assignment or individual. Since the focus is on group project grading, figured I should move on to the actual grading functionality first.

Key parts of the code that could probably benefit from some extra scrutiny:

- Backend stuff (see `backend/src/services/transformers.ts`)
  - Transformation logic 
  - Group sorting algorithm
- Frontend stuff
  - UI feedback as you guys check the dashboard out on your own screens

### CODE REVIEW

**Comment by jwsmith24** [comment]
added the default rating updates into this PR since they're minimal!

**Comment by jwsmith24** [comment]
Andddd just added a couple tweaks to the dashboard layout to clean some stuff up. Done making changes until your review is in!

File: backend/src/services/transformers.ts
Comment (cmccand1) [inline backend/src/services/transformers.ts:1]: Dude, nice doc!

File: docs/gradeSubmissionWorkflow.md
Comment (cmccand1) [inline docs/gradeSubmissionWorkflow.md:1]: Nice doc!

File: frontend/src/features/grading/GradingMain.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { ReactElement, useEffect, useState } from "react";
+import { PaletteAPIResponse, Rubric } from "palette-types";
+import { useFetch } from "@hooks";
+import { useCourse } from "src/context/CourseProvider";
+import { useAssignment } from "../../context/AssignmentProvider.tsx";
+import { LoadingDots } from "../../components/LoadingDots.tsx";
```
Comment (cmccand1) [inline frontend/src/features/grading/GradingMain.tsx:6]: Forget to use `@components` here?

File: frontend/src/features/grading/StatusIcons.tsx
Comment (cmccand1) [inline frontend/src/features/grading/StatusIcons.tsx:1]: Points for a11y here! Cool component too!

File: frontend/src/features/grading/seedData.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+// Define a mock `Criteria` for rubricAssessment
+import { Criteria, Submission } from "palette-types";
+import { GroupedSubmissions } from "palette-types/dist/types/GroupedSubmissions.ts";
```
Comment (cmccand1) [inline frontend/src/features/grading/seedData.ts:3]: Here!

**Review by cmccand1 — CHANGES_REQUESTED** [review]
Looks really good! Only things I could find were pretty superficial

File: frontend/src/features/grading/GradingMain.tsx (hunk 1 above)
Comment (jwsmith24) [reply frontend/src/features/grading/GradingMain.tsx:6]: yes lol

**Review by cmccand1 — APPROVED** [review]
lgtm!

**Pull request merged**